app/src/main/java/org/fossasia/openevent/fragments/LocationsFragment.java

```diff
@@ -124,7 +150,10 @@ public boolean onQueryTextSubmit(String query) {
 
     @Subscribe
     public void RefreshData(RefreshUiEvent event) {
-        locationsListAdapter.refresh();
+
+        if (searchText == null) {
+            locationsListAdapter.refresh();
+        }
```

Comment:
I am not sure if this is right. You are setting `searchText` only 1 time, its not set/reset with every query. But the `RefreshUiEvent ` event can happen any time. There is problem with this.